tools/sqlite3_api_wrapper/sqlite3_udf_api/cast_sqlite.cpp

```diff
@@ -0,0 +1,137 @@
+#include "cast_sqlite.hpp"
+
+/*** Cast Operations *******************************/
+
+// INT casts
+template <>
+sqlite3_value CastSQLite::Operation(int8_t input) {
+	return OperationInt<int8_t>(input);
+}
+
+template <>
+sqlite3_value CastSQLite::Operation(int16_t input) {
+	return OperationInt<int16_t>(input);
+}
+
+template <>
+sqlite3_value CastSQLite::Operation(int32_t input) {
+	return OperationInt<int32_t>(input);
+}
+
+template <>
+sqlite3_value CastSQLite::Operation(int64_t input) {
+	return OperationInt<int64_t>(input);
+}
+
+// FLOAT casts
+template <>
+sqlite3_value CastSQLite::Operation(float input) {
+	return OperationFloat<float>(input);
+}
+
+template <>
+sqlite3_value CastSQLite::Operation(double input) {
+	return OperationFloat<double>(input);
+}
+
+// STRING cast
+template <>
+sqlite3_value CastSQLite::Operation(string_t input) {
+	sqlite3_value sqlite_str;
+	sqlite_str.type = SQLiteTypeValue::TEXT;
+	sqlite_str.n = input.GetSize();
+	sqlite_str.str_t = input;
+	return sqlite_str;
+}
+
+sqlite3_value CastSQLite::OperationBlob(string_t input) {
+	sqlite3_value sqlite_blob;
+	sqlite_blob.type = SQLiteTypeValue::BLOB;
+	sqlite_blob.n = input.GetSize();
+	sqlite_blob.str_t = input;
+	return sqlite_blob;
+}
+/*** Get Values ***********************************/
+template <>
+int64_t CastSQLite::GetValue(sqlite3_value input) {
+	return input.u.i;
+}
+
+template <>
+double CastSQLite::GetValue(sqlite3_value input) {
+	return input.u.r;
+}
+
+template <>
+string_t CastSQLite::GetValue(sqlite3_value input) {
+	return input.str_t;
+}
+
+//*** Cast to vectors ***********************************/
+unique_ptr<vector<sqlite3_value>> CastSQLite::ToVectorSQLite(LogicalType type, VectorData &vec_data, idx_t size) {
+	LogicalTypeId type_id = type.id();
+	switch (type_id) {
+	case LogicalTypeId::TINYINT: {
```

Comment:
LogicalType::NULL is missing. Could you add a test in the form of `SELECT my_sqlite_function(NULL)`?